streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -288,6 +291,8 @@ public StreamThread(TopologyBuilder builder,
         this.standbyRecords = new HashMap<>();
 
         this.stateDirectory = new StateDirectory(applicationId, config.getString(StreamsConfig.STATE_DIR_CONFIG), time);
+        Object maxPollInterval = consumerConfigs.get(ConsumerConfig.MAX_POLL_INTERVAL_MS_CONFIG);
+        this.rebalanceTimeoutMs =  maxPollInterval == null ? 300000 : (Integer) ConfigDef.parseType(ConsumerConfig.MAX_POLL_INTERVAL_MS_CONFIG, maxPollInterval, Type.INT);
```

Comment:
@sjmittal You are right, I misinterpreted the line here.

I agree with @dguy that we could add this into `CONSUMER_DEFAULT_OVERRIDES` inside `StreamsConfig`. Note that it will only take effect if users do NOT specify these configs themselves.